### DISCUSSION TITLE
history in non-interactive mode

### DISCUSSION THREAD

**Discussion by indiscible** [xonsh/xonsh#5348] [Q&A] [answered]
i would like to run a xsh script and then inspect the history of shell commands with their timestamps, stdout, stderr etc..
it seems history generation is deactivated if you don't run in interactive mode.

**Comment by anki-code** [comment, marked as answer]
Hey! Could you please describe your goal?
What about using [`$XONSH_TRACE_SUBPROC`](https://xon.sh/envvars.html#xonsh-trace-subproc)?

**Reply by indiscible** [reply]
hey, let say I have a script that run multiple commands (checkout, build, run multiple tests) and that script is run automatically.
That'd be cool to use the history feature to get a report/audit of how each command ran: duration, stdout, stderr ...

**Reply by anki-code** [reply]
If you want to catch errors, take a look into https://xon.sh/envvars.html#raise-subproc-error

History was made for interactive xonsh session where the commands separated by user. In non-interactive session history is not working.

If you understand how it should work for non-interactive PR is welcome!
